terraform/resource_address.go

```diff
@@ -329,6 +329,58 @@ func (addr *ResourceAddress) Equals(raw interface{}) bool {
 		modeMatch
 }
 
+// Less returns true if and only if the receiver should be sorted before
+// the given address when presenting a list of resource addresses to
+// an end-user.
+//
+// This sort uses lexicographic sorting for most components, but uses
+// numeric sort for indices, thus causing index 10 to sort after
+// index 9, rather than after index 1.
+func (addr *ResourceAddress) Less(other *ResourceAddress) bool {
+
+	switch {
+
+	case len(addr.Path) < len(other.Path):
+		return true
+
+	case !reflect.DeepEqual(addr.Path, other.Path):
```

Comment:
I don't think it is right now, but is it possible for indexes to end up in a path? Might they be if we can add `count` support to modules? 

This is probably fine for now, I'm only mentioning it because I think I've fixed "lexically sorted numbers" 2 or 3 times in the past couple releases.